### PR TITLE
dev-python/ipython: Fix security issue

### PR DESCRIPTION
**Pull request by marbre**
"Maliciously crafted text files in IPython/Jupyter editor"
http://seclists.org/oss-sec/2015/q3/558
https://bugs.gentoo.org/show_bug.cgi?id=560708

Package-Manager: portage-2.2.20.1

### CODE REVIEW

**Comment by marbre** [comment]
Patches are created with `git format-patch`, they apply without error and ipython can still be installed. Beside, `test` fails for ipython-3.2.1 (unpatches) and thus is does for ipython-3.2.1-r1 (patches). This should solve the security issue, since only a very few commits were made after the release of ipython-3.2.1. Didn't tested explicitly for the vulnerability. 

Patches also apply to ipython-3.2.0.

File: dev-python/ipython/ipython-3.2.1-r1.ebuild [hunk 1]
```diff
@@ -15,7 +15,7 @@ SRC_URI="mirror://pypi/${PN:0:1}/${PN}/${P}.tar.gz"
 
 LICENSE="BSD"
 SLOT="0"
-KEYWORDS="amd64 ~arm ~mips ppc ppc64 x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos"
+KEYWORDS="~amd64 ~arm ~mips ~ppc ~ppc64 ~x86 ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~x86-macos"
```
Comment (jlec) [inline dev-python/ipython/ipython-3.2.1-r1.ebuild]: You cannot move and change keywords. Please copy the ebuild.

**Comment by jlec** [comment]
Is there a simply way to fix the tests? Any idea?

File: dev-python/ipython/ipython-3.2.1-r1.ebuild (hunk 1 above)
Comment (marbre) [inline dev-python/ipython/ipython-3.2.1-r1.ebuild]: "When upgrading, drop all existing keywords from arch to ~arch, and leave any existing ~arch keywords intact. [..] This also applies to revision bumps, not just to upstream version changes." [[1](https://devmanual.gentoo.org/keywording/index.html#keywording-on-upgrades)]

Or do you mean I should leave the `ipython-3.2.1.ebuild` and add an additional `ipython-3.2.1-r1.ebuild`?

**Comment by marbre** [comment]
Just ran the tests on another machine. No issues with 3.2.1 and 3.2.1-r1. Sorry for the inconveniences.

**Comment by marbre** [comment]
@jlec Leaving 3.2.1 untouched, adding an additional 3.2.1-r1.

**Pull request merged**